docs/proposals/002-api-proposal/proposal.md

```diff
@@ -80,101 +80,99 @@ The API design is based on these axioms:
-        // The name should be unique among the services that reference the same backend pool.
+        // The name should be unique among the workloads that reference the same backend pool.
         // This is the parameter that will be used to match the request with. In the future, we may
         // allow to match on other request parameters. The other approach to support matching on 
         // on other request parameters is to use a different ModelName per HTTPFilter.
         // Names can be reserved without implementing an actual model in the pool.
         // This can be done by specifying a target model and setting the weight to zero,
         // an error will be returned specifying that no valid target model is found.
-        Name string
+        ModelName string
         // Optional
-        // LLM Services with an objective have higher priority than services without.
-        // IMPORTANT: By specifying an objective, this places the LLMService in a higher priority class than LLMServices without a defined priority class. 
-        // In the face of resource-scarcity. Higher priority requests will be preserved, and lower priority class requests will be rejected.
-        Objective *Objective
+        // Defines how important it is to serve the model compared to other models referencing the same pool.
+        Criticality *Criticality
         // Optional.
-	      // Allow multiple versions of a model for traffic splitting. 
-	      // If not specified, the target model name is defaulted to the modelName parameter.
+	    // Allow multiple versions of a model for traffic splitting. 
+	    // If not specified, the target model name is defaulted to the modelName parameter.
         // modelName is often in reference to a LoRA adapter.
         TargetModels []TargetModel
+        // Reference to the InferencePool that the model registers to. It must exist in the same namespace.
+        PoolRef InferencePoolReference
 }
 
-
+// Defines how important it is to serve the model compared to other models.
+type Criticality string
+const (
+    // Most important. Requests to this band will be shed last.
+    Critical  Criticality = "Critical"
+    // More important than Sheddable, less important than Critical.
+    // Requests in this band will be shed before critical traffic.
+    Default  Criticality = "Default"
+    // Least important. Requests to this band will be shed before all other bands.
+    Sheddable  Criticality = "Sheddable"
+ )
 
 // TargetModel represents a deployed model or a LoRA adapter. The
 // Name field is expected to match the name of the LoRA adapter
 // (or base model) as it is registered within the model server. Inference
 // Gateway assumes that the model exists on the model server and is the
 // responsibility of the user to validate a correct match. Should a model fail
 // to exist at request time, the error is processed by the Instance Gateway,
```

Comment:
We should unify the term, here we use "Instance Gateway", on line 171 we use "Inference Gateway", and at the beginning we used "LLM Instance Gateway"

Reply:
updated throughout